render/topologies.go

```diff
@@ -25,6 +29,45 @@ var ProcessRenderer = MakeReduce(
 	},
 )
 
+// ProcessWithContainerNameRenderer is a Renderer which produces a process
+// graph enriched with container names where appropriate
+type ProcessWithContainerNameRenderer struct{}
+
+// Render produces a process graph where the minor labels contain the
+// container name, if found.
+func (r ProcessWithContainerNameRenderer) Render(rpt report.Report) RenderableNodes {
+	var processes = ProcessRenderer.Render(rpt)
+	var containers = LeafMap{
+		Selector: report.SelectContainer,
+		Mapper:   MapContainerIdentity,
+		Pseudo:   PanicPseudoNode,
+	}.Render(rpt)
+
+	for id, p := range processes {
+		pid, ok := p.NodeMetadata.Metadata[process.PID]
+		if !ok {
+			continue
+		}
+		containerID, ok := p.NodeMetadata.Metadata[docker.ContainerID]
+		if !ok {
+			continue
+		}
+		container, ok := containers[containerID]
+		if !ok {
+			continue
+		}
+		p.LabelMinor = fmt.Sprintf("%s (%s:%s)", report.ExtractHostID(p.NodeMetadata), container.LabelMajor, pid)
```

Comment:
Just to be clear, this is overwriting the existing minor label, which is of the format "hostname (pid)" — right?